### PR TITLE
Fix Code Coverage Showing Invalid Values with GraphQL Compiler Plugin

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

$Subject

Fixes: [#6418](https://github.com/ballerina-platform/ballerina-library/issues/6418)
Fixes: [#6310](https://github.com/ballerina-platform/ballerina-library/issues/6310)

## Examples

N/A

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] ~Updated the spec~
- [x] Checked native-image compatibility
- [ ] No `commons` package changes (if there are any, please update the GraphQL version in [GraphQL tools](https://github.com/ballerina-platform/graphql-tools) and [Ballerina dev tools](https://github.com/ballerina-platform/ballerina-dev-tools))
- [ ] No `compiler` package changes (if there are any, please update the GraphQL version in [Ballerina dev tools](https://github.com/ballerina-platform/ballerina-dev-tools))

### CODE REVIEW

**Review by DimuthuMadushan — APPROVED** [review]

File: compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/GraphqlSourceModifier.java [hunk 1]
```diff
@@ -445,10 +443,13 @@ private SeparatedNodeList<MappingFieldNode> getUpdatedFields(MappingConstructorE
...
-        for (MappingFieldNode field : existingFields) {
-            fields.add(field);
-            fields.add(separator);
+        for (int i = 0; i < existingFields.size(); i++) {
+            fields.add(existingFields.get(i));
+            if (i < existingFields.size() - 1) {
+                fields.add(existingFields.getSeparator(i));
+            }
         }
+        fields.add(separator);
```
Comment (SasinduDilshara) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/GraphqlSourceModifier.java:452]: if `existingFields.size()` is zero, then this separator will add in the first ryt?

File: compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/GraphqlSourceModifier.java [hunk 2]
```diff
@@ -445,10 +443,13 @@ private SeparatedNodeList<MappingFieldNode> getUpdatedFields(MappingConstructorE
         List<Node> fields = new ArrayList<>();
         SeparatedNodeList<MappingFieldNode> existingFields = annotationValue.fields();
         Token separator = NodeFactory.createToken(SyntaxKind.COMMA_TOKEN);
-        for (MappingFieldNode field : existingFields) {
-            fields.add(field);
-            fields.add(separator);
+        for (int i = 0; i < existingFields.size(); i++) {
+            fields.add(existingFields.get(i));
+            if (i < existingFields.size() - 1) {
```
Comment (SasinduDilshara) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/GraphqlSourceModifier.java:448]: I think, if the seize is 2, we need to add a separator in here.

File: compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/GraphqlSourceModifier.java (hunk 1 above)
Comment (ThisaruGuruge) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/GraphqlSourceModifier.java:452]: Having `0` fields in the annotation does not make sense, right?

File: compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/GraphqlSourceModifier.java (hunk 2 above)
Comment (ThisaruGuruge) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/GraphqlSourceModifier.java:448]: Sorry, I did not understand this point.

**Review by SasinduDilshara — APPROVED** [review]

**Pull request merged**